core/src/main/java/com/graphhopper/storage/index/LocationIndexTree.java

```diff
@@ -254,10 +254,6 @@ public boolean loadExisting() {
         if (dataAccess.getHeader(0) != MAGIC_INT)
             throw new IllegalStateException("incorrect location index version, expected:" + MAGIC_INT);
 
-        if (dataAccess.getHeader(1 * 4) != calcChecksum())
-            throw new IllegalStateException("location index was opened with incorrect graph: "
-                    + dataAccess.getHeader(1 * 4) + " vs. " + calcChecksum());
```

Comment:
This check is necessary in rare cases when GH crashes somewhere in the middle of a CH or LM preparation I think and no graph files are on disk yet (but the loc index is). If then a new import with a different PBF is done it will load the wrong LocationIndex files from disk.

We could try to flush the LocationIndex only at the end on success. But we don't do this at the moment, because we flush files as soon as they are ready to avoid e.g. a re-import of the PBF file if only the CH preparation failed. This was sometimes handy for debugging CH or LM for larger areas, but don't know if it is worth the trouble.

Another solution could be to:

 1. call basegraph.freeze before LocationIndex (and ensure basegraph.freeze was called) 
 2. write the basgraph node+edge count into the header of the `nodes` DataAccess (on basegraph.freeze)
 3. use this basegraph header information to get the checksum (the problem is that we would need to use a bit ugly code to get this header info `dir.find("nodes").getHeader()` instead of using the `Graph` interface)